torch/distributed/_tensor/README.md

```diff
@@ -1,3 +1,167 @@
-# Distributed Tensor
+# PyTorch DistributedTensor (DTensor)
 
-This is a prototype distributed tensor implementation that implements most of the basic parts in the RFC https://docs.google.com/document/d/15R3fmoPbzedlKSjtpQ97HFPidp9QTXLEap6gyIvRrMY/edit#
+This folder contains the DistributedTensor (a.k.a DTensor) implementation in PyTorch.
+
+## Introduction
+We propose distributed tensor primitives to allow easier distributed computation authoring in SPMD(Single Program Multiple Devices) paradigm. The primitives are simple but powerful when used to express tensor distributions with both sharding and replication parallelism strategies. This could empower native Tensor parallelism among other advanced parallelism explorations. For example, to shard a big tensor across devices with 3 lines of code:
+
+```python
+import torch
+from torch.distributed import DeviceMesh, Shard, distribute_tensor
+
+# Create a mesh topology with the available devices.
+mesh = DeviceMesh("cuda", list(range(world_size)))
+big_tensor = torch.randn(100000, 88)
+# Shard this tensor over the mesh by sharding `big_tensor`'s 0th dimension over the 0th dimension of `mesh`.
+my_dtensor = distribute_tensor(big_tensor, mesh, [Shard(dim=0)])
+```
+
+## Motivation
+
+Today there are mainly three ways to scale up distributed training: Data Parallel, Tensor Parallel and Pipeline Parallel. Each of them works on a separate dimension where solutions have been built independently (i.e. PyTorch DDP, FSDP, ShardedTensor, PiPPy, etc.). When training really large models, users would like to use these technologies together (i.e. 3-D Parallelism), while the interoperability of the existing solutions are not great and often hard to use (i.e. users might want arbitrary combinations of the data parallel, tensor parallel and pipeline parallel). This is becoming an issue for users and one of the biggest reasons is that there’s no common abstractions that build the bridge between different parallelism strategies.
+
+An ideal scenario is that users could just build their models like in a single node/device, without worrying about how to do distributed training in a cluster, and our solutions could help them run distributed training in an efficient manner. For example, researchers just need to build their big transformer model, and PyTorch Distributed automatically figures out how to split the model and run pipeline parallel across different nodes, how to run data parallel and tensor parallel within each node. In order to achieve this, we need some common abstractions to represent data distribution and run the distributed computation.
```

Comment:
nit:

"An ideal scenario is that users could just build their models like in a single node/device" -> "An ideal scenario is that users could build their distributed program just like authoring a single-node/device one"

Reply:
"data distribution" sometimes refers to different things, e.g., Gaussian distribution. How about "we need some common abstractions to distribute tensor values and distributed computations accordingly."